### DISCUSSION TITLE
estimate clickhouse query cost

### DISCUSSION THREAD

**Discussion by athree** [ClickHouse/ClickHouse#49058] [Q&A] [answered]
is there any functions or tools based on clickhouse can estimate query cost, such as cpu/mem?

while i can only get estimated rows、marks、parts by explain estimate ${SQL}.

**Comment by athree** [comment]
i means estimate query cost before query excuted, not logged in system.processes or  system.query_log...

**Comment by alexey-milovidov** [comment, marked as answer]
No, there are no such tools.

**Reply by athree** [reply]
got， and any plan to do support this estimate?

i seems no plan in Roadmap 2023.

**Reply by alexey-milovidov** [reply]
Not currently.

**Reply by athree** [reply]
ok, thanks

**Reply by AhmadDarwich** [reply]
Hello, built a small CLI :chqce estimates a query's cost before you run it — estimated rows/parts/marks from EXPLAIN ESTIMATE, client + server timing, rows/bytes read, peak memory, and it suggests skip-indexes by comparing your WHERE columns against the table's ORDER BY. [clickhouse-query-cost-estimator](https://github.com/AhmadDarwich/clickhouse-query-cost-estimator).
